DESIGN.md

```diff
@@ -0,0 +1,73 @@
+# Extensions
+
+*Note*: This document is evolving and is in draft state.
+
+Plugin architecture enables extending core features of OpenSearch. There are various kinds of plugins which are supported.
+But, the architecture has significant problems for OpenSearch customers. Importantly, plugins can fatally impact the cluster
+i.e critical workloads like ingestion/search traffic would be impacted because of a non-critical plugin like s3-repository failed with an exception.
+
+This problem is exponentially grows when we would like to run a 3rd Party plugin from the community.  
+As OpenSearch and plugins run in the same process, it brings in security risk, dependency conflicts and reduces the velocity of releases.
+
+Introducing extensions, a simple and easy way to extend features of OpenSearch. It would support all plugin features and enable them to run in a different process or another node via OpenSearch SDK.
```

Comment:
```suggestion
Introducing extensions, a simple and easy way to extend features of OpenSearch. It would support all plugin features and enable them to run in a separate process or on another node via OpenSearch SDK.
```